core/math/vector3.h

```diff
@@ -223,10 +223,15 @@ Vector3 Vector3::slerp(const Vector3 &p_to, const real_t p_weight) const {
 		// Zero length vectors have no angle, so the best we can do is either lerp or throw an error.
 		return lerp(p_to, p_weight);
 	}
+	Vector3 axis = cross(p_to).normalized();
+	if (unlikely(axis == Vector3())) {
+		// Colinear vectors have no rotation axis or angle between them, so the best we can do is lerp.
+		return lerp(p_to, p_weight);
+	}
```

Comment:
This will be faster to code longhand the normalize function so you can check the length directly (or square length) rather than checking 3 floats. Incidentally this is one of the reasons why the technique in #56567 is so useful.

Reply:
Great idea, I've updated the PR to have `normalized` be longhand.